# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsBaseDragService.h
# Commit: beeda5ef3712
# Full Hash: beeda5ef3712b73a19c7a4ce08a52ab4dd08b802
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 20 - Move EndDragSession from nsIDragService to nsIDragSession r=gstoll,rkraesig,win-reviewers,geckoview-reviewers,m_kato
#   
#   This is a straightforward move.  It does take the liberty of breaking out an
#   EndDragSessionImpl method, which will be needed later.
#   
# ==============================================================================

diff -r 6c3ffc8c534c -r beeda5ef3712 widget/nsBaseDragService.h
--- a/widget/nsBaseDragService.h	Thu Jul 04 07:48:10 2024 +0000
+++ b/widget/nsBaseDragService.h	Thu Jul 04 07:48:11 2024 +0000
@@ -72,6 +72,9 @@
   nsBaseDragSession();
   ~nsBaseDragSession();
 
+  MOZ_CAN_RUN_SCRIPT virtual nsresult EndDragSessionImpl(
+      bool aDoneDrag, uint32_t aKeyModifiers);
+
   // Returns true if a drag event was dispatched to a child process after
   // the previous TakeDragEventDispatchedToChildProcess() call.
   bool TakeDragEventDispatchedToChildProcess() {
@@ -117,6 +120,11 @@
 
   // Weak references to PBrowsers that are currently engaged in drags
   nsTArray<nsWeakPtr> mBrowsers;
+  // remote drag data
+  RefPtr<mozilla::dom::RemoteDragStartData> mDragStartData;
+
+  // Sub-region for tree-selections.
+  mozilla::Maybe<mozilla::CSSIntRegion> mRegion;
 
   // the screen position where drag gesture occurred, used for positioning the
   // drag image.
@@ -149,6 +157,11 @@
 
   bool mIsDraggingTextInTextControl = false;
   bool mSessionIsSynthesizedForTests = false;
+
+  // true if in EndDragSession
+  bool mEndingSession = false;
+  // true if mImage should be used to set a drag image
+  bool mHasImage = false;
 };
 
 /**
@@ -250,18 +263,10 @@
 
   virtual bool IsMockService() { return false; }
 
-  // true if in EndDragSession
-  bool mEndingSession;
-  // true if mImage should be used to set a drag image
-  bool mHasImage;
-
   // the contentpolicy type passed to the channel when initiating the drag
   // session
   nsContentPolicyType mContentPolicyType;
 
-  // remote drag data
-  RefPtr<mozilla::dom::RemoteDragStartData> mDragStartData;
-
   uint32_t mSuppressLevel;
 
   // Sub-region for tree-selections.